core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1793,7 +1793,7 @@ final class Stream[+F[_], +O] private[fs2] (private val free: FreeC[F, O, Unit])
     * }}}
     */
   def map[O2](f: O => O2): Stream[F, O2] =
-    this.pull.echo.mapOutput(f).stream
+    this.pull.echo.mapOutput(f).streamNoScope
```

Comment:
Without this change, the test for `resourceWeak` fails b/c there's a `map` in the implementation of `resourceWeak`. It's okay to not have a scope here b/c this pull consumes the entire source stream. Note back in 1.0.5, we had this same definition of `map` and it was the only such invocation in the library. So while this does feel a bit ad-hoc, we also have the most mileage on it.